src/probnum/utils/linalg/_inner_product.py

```diff
@@ -0,0 +1,65 @@
+"""Functions defining useful inner products."""
+
+from typing import Optional, Union
+
+import numpy as np
+
+from probnum import linops
+
+
```

Comment:
This is a stylistic comment. Typically "euclidean" norm and inner product only refers to the 2-norm and the inner product that induces it.

Reply:
How about we just call these functions `inprod` and `induced_norm`, since every (semi-)inner product on R^n can be represented by a positive (semi-)definite matrix `A`?

Reply:
That sounds good to me